src/Sylius/Component/Core/Statistics/Chart/ChartFactory.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Core\Statistics\Chart;
+
+use Webmozart\Assert\Assert;
+
+final class ChartFactory implements ChartFactoryInterface
```

Comment:
I wonder if we need this service, are we not able to get this data from the already existed and used in UI service `SalesDataProvider` 🤔

Reply:
The signature of `getSalesSummary` is not convenient to work with. Currently, we are using a \DatePeriod object to pass the necessary data and returning a different DTO (`SalesSummaryInterface`). I believe that introducing a new service to clarify naming and simplify the code would be necessary.